src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -51,78 +56,129 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
       getActiveCluster(googleProject, clusterName, dataAccess)
     }
   }
 
-  def deleteCluster(googleProject: GoogleProject, clusterName: ClusterName): Future[Int] = {
-    getActiveClusterDetails(googleProject, clusterName) flatMap { cluster =>
-      if(cluster.status.isActive) {
-        // Delete the service account key in Google, if present
-        val deleteServiceAccountKey = dbRef.inTransaction { dataAccess =>
-          dataAccess.clusterQuery.getServiceAccountKeyId(googleProject, clusterName)
-        } flatMap {
-          case Some(key) => googleIamDAO.removeServiceAccountKey(dataprocConfig.leoGoogleProject, cluster.googleServiceAccount, key)
-          case None => Future.successful(())
-        }
+  def deleteCluster(userInfo: UserInfo, googleProject: GoogleProject, clusterName: ClusterName): Future[Int] = {
+    for {
+      //throws 404 if no permissions
+      cluster <- getActiveClusterDetails(userInfo, googleProject, clusterName)
 
-        for {
-          _ <- deleteServiceAccountKey
-          _ <- gdDAO.deleteCluster(googleProject, clusterName)
-          recordCount <- dbRef.inTransaction(dataAccess => dataAccess.clusterQuery.markPendingDeletion(cluster.googleId))
-        } yield {
-          clusterMonitorSupervisor ! ClusterDeleted(cluster)
-          recordCount
-        }
-      } else Future.successful(0)
-    }
+      //if you've got to here you at least have GetClusterDetails permissions so a 401 is appropriate if you can't actually destroy it
+      _ <- checkClusterPermission(userInfo,  DeleteCluster, cluster, throw401 = true)
+
+      count <- internalDeleteCluster(cluster)
+    } yield { count }
   }
 
-  def listClusters(params: LabelMap): Future[Seq[Cluster]] = {
-   processListClustersParameters(params).flatMap { paramMap =>
-     dbRef.inTransaction { dataAccess =>
-       dataAccess.clusterQuery.listByLabels(paramMap._1, paramMap._2)
-     }
-   }
+  def internalDeleteCluster(cluster: Cluster): Future[Int] = {
+    if(cluster.status.isActive) {
+      // Delete the service account key in Google, if present
+      val deleteServiceAccountKey = dbRef.inTransaction { dataAccess =>
+        dataAccess.clusterQuery.getServiceAccountKeyId(cluster.googleProject, cluster.clusterName)
+      } flatMap {
+        case Some(key) => googleIamDAO.removeServiceAccountKey(dataprocConfig.leoGoogleProject, cluster.googleServiceAccount, key)
+        case None => Future.successful(())
+      }
+
+      for {
+        _ <- deleteServiceAccountKey
+        _ <- gdDAO.deleteCluster(cluster.googleProject, cluster.clusterName)
+        recordCount <- dbRef.inTransaction(dataAccess => dataAccess.clusterQuery.markPendingDeletion(cluster.googleId))
+        _ <- authProvider.notifyClusterCreated(cluster.creator.value, cluster.googleProject.value, cluster.googleId)
+      } yield {
+        clusterMonitorSupervisor ! ClusterDeleted(cluster)
+        recordCount
+      }
+    } else Future.successful(0)
```

Comment:
I feel like this should return `Future[Unit]`, I don't think anyone is using the `Int`